### PR TITLE
Jsf 1386 create deploy reusable workflow

### PR DESCRIPTION
**Pull request by hyperwidget**
Create a reusable workflow to deploy to S3.
Updates the README with details about this workflow

You can find a successful run of this workflow (against this branch, specifically) [here](https://github.com/janeapp/react-app-starter-kit/actions/runs/3651919702/jobs/6169699246) and can find the S3 bucket that it successfully deployed to [here](https://s3.console.aws.amazon.com/s3/buckets/react-starter-kit-bucket?region=us-west-2&tab=objects)

[Linear Ticket](https://linear.app/jane/issue/JSF-1386/create-deploy-reusable-workflow)

### CODE REVIEW

File: .github/workflows/deploy-frontend-app.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        run: |
+          unzip build.zip
+      - name: setup AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.AWS_ROLE_TO_ASSUME }}
+          aws-region: ${{ secrets.AWS_REGION }}
+      - name: Copy files to S3
+        run: |
+          aws s3 mv build s3://${{ secrets.AWS_S3_BUCKET_NAME }}/build --recursive
```
Comment (carvallegro) [inline .github/workflows/deploy-frontend-app.yml:39]: To confirm with @alonso-jane but I don't think the [Terraform code for Cloudfront](https://github.com/janeapp/terraform-spa-app/blob/483f977cd661429cf167e5013866267b1100d504/terraform/modules/spa-cdn/cloudfront.tf#L19) is configured to link to `build/` by default. We might need to either:

- Change this code to copy the files to the root of the bucket
- Change the Terraform code to point to `build` by default.

File: README.md [hunk 2]
```diff
@@ -21,10 +21,30 @@ This reusable workflow is here to ease the build and release process of React ap
...
+`AWS_ROLE_TO_ASSUME`: The role created through terraform that this workflow should use to deploy to S3
+`AWS_S3_BUCKET_NAME`: The name of the bucket created with the repository's terraform script
+
+#### Inputs
+
+| Name           | Default value | Description                                                                 |
+| -------------- | ------------- | --------------------------------------------------------------------------- |
+| version_number | na            | Version number of the release in github to deploy                           |
+| environment    | na            | Github environment to get secrets from, the environment you're deploying to |
+|  |
```
Comment (carvallegro) [inline README.md:50]: Could you remove this line? It creates an empty table line otherwise

File: README.md [hunk 3]
```diff
@@ -21,10 +21,30 @@ This reusable workflow is here to ease the build and release process of React ap
...
+
+### Deploy frontend app
+
+This reusable workflow is here to ease the deployment process of a built and released React applications to S3.
+
+#### Secrets
+
+These secrets should all be inherited by the calling repository and are typically created through terraform
+
+`AWS_REGION`: The region of the calling repository's S3 bucket
```
Comment (carvallegro) [inline README.md:40]: Should we suggest a default region? We need it to setup AWS but we are using S3 which is a global resource so we could have a default value

File: .github/workflows/deploy-frontend-app.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+name: Deploy release to S3
+
+on:
+  workflow_call:
```
Comment (carvallegro) [inline .github/workflows/deploy-frontend-app.yml:4]: Could you also declare the secrets in this block? It's prescribed by [the documentation](https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-inputs-and-secrets-in-a-reusable-workflow). 

I think we can set `AWS_REGION` as not required

File: README.md (hunk 3 above)
Comment (hyperwidget) [reply README.md]: I added a blurb mentioning that the default value of `aws-west-2` can be used, as well as updated the script to have that as a default

File: README.md [hunk 5]
```diff
@@ -21,10 +21,29 @@ This reusable workflow is here to ease the build and release process of React ap
...
+
+This reusable workflow is here to ease the deployment process of a built and released React applications to S3.
+
+#### Secrets
+
+These secrets should all be inherited by the calling repository and are typically created through terraform
+
+`AWS_REGION`: The region of the calling repository's S3 bucket; necessary to deploy to S3, but can use the default value of `us-west-2`
+`AWS_ROLE_TO_ASSUME`: The role created through terraform that this workflow should use to deploy to S3
+`AWS_S3_BUCKET_NAME`: The name of the bucket created with the repository's terraform script
```
Comment (tavisatjane) [inline README.md:42]:
```suggestion
* `AWS_REGION`: The region of the calling repository's S3 bucket; necessary to deploy to S3, but can use the default value of `us-west-2`
* `AWS_ROLE_TO_ASSUME`: The role created through terraform that this workflow should use to deploy to S3
* `AWS_S3_BUCKET_NAME`: The name of the bucket created with the repository's terraform script
```

**Review by tavisatjane — APPROVED** [review]

**Review by carvallegro — APPROVED** [review]

**Pull request merged**